src/main/java/prefs/Preferences.java

```diff
@@ -128,15 +138,32 @@ public void setLastOpenBoard(String board) {
     }
     
     public Optional<String> switchBoard() {
-        if (getLastOpenBoard().isPresent() && getAllBoards().size() > 1) {
-            List<String> boardNames = getAllBoardNames();
+        setBoardToSwitch();
+        return getLastOpenBoard();
+    }
+
+    /**
+     * Sets the lastOpenBoard to the board to switch to
+     */
+    private void setBoardToSwitch(){
```

Comment:
Yes, we can delay a implementation while conflicting PRs get through (hopefully, respective devs will push them through quickly).
By 'serialize' you mean convert to json right?